src/vm/eventtrace.cpp

```diff
@@ -6805,15 +6805,34 @@ VOID ETW::MethodLog::SendEventsForNgenMethods(Module *pModule, DWORD dwEventOpti
     } CONTRACTL_END;
 
 #ifdef FEATURE_PREJIT
-    if(!pModule || !pModule->HasNativeImage())
+    if (!pModule)
         return;
 
-    MethodIterator mi(pModule);
+    PEImageLayout * pLoadedLayout = pModule->GetFile()->GetLoaded();
+    _ASSERTE(pLoadedLayout != NULL);
 
-    while(mi.Next())
+#ifdef FEATURE_READYTORUN
+    if (pLoadedLayout->HasReadyToRunHeader())
     {
-        MethodDesc *hotDesc = (MethodDesc *)mi.GetMethodDesc();
-        ETW::MethodLog::SendMethodEvent(hotDesc, dwEventOptions, FALSE);
+        ReadyToRunInfo::MethodIterator mi(pModule->GetReadyToRunInfo());
+        while (mi.Next())
+        {
+            MethodDesc *hotDesc = (MethodDesc *)mi.GetMethodDesc();
```

Comment:
I think this is very bug prone because of it will cause loader to run. There is similar code on the codepath for fragile ngen images - it is disabled with the following comment:

```
        // Forcibly restoring ngen methods can cause all sorts of deadlocks and contract violations
        // These events are therefore put under the private provider
```

Instead, as a quick fix - only report method that has methoddesc already: lookup the entrypoint in `ReadyToRunInfo::m_entryPointToMethodDescMap` and skip it if it is not there.